lib/puppet/type/wls_server/arguments.rb

```diff
@@ -1,24 +1,45 @@
-newproperty(:arguments) do
+newproperty(:arguments, :array_matching => :all) do
   include EasyType
 
-  desc 'The server arguments of the server'
+  desc 'The server sttart arguments of the server.
+    Can be specified as an array or a space separated string'
+  #or both...
+  #The following should also work
+  #arguments => ['arg1 arg2','arg3','arg4 arg5']
+  #and be idempotent with 'arg4 arg3 arg1 arg2 arg5' and [arg1,arg4,arg2,'arg3 arg5'] etc
 
   to_translate_to_resource do | raw_resource|
-    return '' if raw_resource['arguments'].nil?
-    raw_resource['arguments']
+    return [] if raw_resource['arguments'].nil?
+    raw_resource['arguments'].split(/\s/)
   end
 
-  def should=(values)
-    values = values.flatten.join("\n") if values.is_a?(Array)
-    @should = values
+  munge do |value|
+    if value.is_a?(String)
```

Comment:
Since we still support space separate string arguments, we need to munge these to arrays...

But, this confused me for a while.  value is never actually an array.  If arguments is specified as an array, munge is called multiple times for each array element...